### PR TITLE
AVC: Added initial field picture decode support.

### PR DESCRIPTION
**Pull request by jeffqjiangNew**
- 17 conformance streams pass.

### CODE REVIEW

File: src/parser/avc_parser.cpp [hunk 1]
```diff
@@ -192,6 +193,11 @@ ParserResult AvcVideoParser::ParsePictureData(const uint8_t *p_stream, uint32_t
 
                     // Start decode process
                     if (num_slices_ == 0) {
+                        if (p_slice_header->field_pic_flag) {
+                            second_field_ = (pic_count_ & 1) ? 1 : 0;
```
Comment (rrawther) [inline src/parser/avc_parser.cpp]: second_field = (pic_count_ & 1); will be sufficient here. Also if we move line #199 to #196, else condition can be avoided

**Review by rrawther — COMMENTED** [review]
Added minor comment

File: src/parser/avc_parser.cpp (hunk 1 above)
Comment (jeffqjiangNew) [reply src/parser/avc_parser.cpp]: Yes. :) Simplified line 197. 

We need to keep line 199 for frame pictures (set to 0). second_filed_ value is checked later.

File: src/parser/avc_parser.cpp [hunk 2]
```diff
@@ -1997,44 +2033,82 @@ static inline int CompareLongTermPicNumAsc(const void *p_pic_info_1, const void
     }
 }
 
+static inline int CompareFrameNumWrapDesc(const void *p_pic_info_1, const void *p_pic_info_2) {
+    int frame_num_wrap_1 = ((AvcVideoParser::AvcPicture*)p_pic_info_1)->frame_num_wrap;
```
Comment (AryanSalmanpour) [inline src/parser/avc_parser.cpp]: please use c++ style cast (reinterpret_cast) for p_pic_info_1 and p_pic_info_2 on lines 2037-2038 and 2050-2051.

**Review by AryanSalmanpour — CHANGES_REQUESTED** [review]

File: src/parser/avc_parser.cpp (hunk 2 above)
Comment (jeffqjiangNew) [reply src/parser/avc_parser.cpp]: Done.

**Review by AryanSalmanpour — APPROVED** [review]

**Pull request merged**